roles/ceph-monitor/files/ceph_adjust.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/python
+# coding: utf-8 -*-
+
+# ceph wants pg number be power of 2(good for cursh).
+# So if we increase pg number from 1024, the next number will be 2048.
+# Increasing so many pg numbers will cause much data rebalancing.
+# This script is to increase pg number in small step,
+# we run this cript as cron job every 5 minutes.
+
+import subprocess
+import pickle
+import logging
+import json
+import os
+import time
+import re
+import glob
+
+PKLFILE = '/var/lib/ceph_adjust/pgsonpool.pkl'
+LOGFILE = '/var/log/ceph/ceph_adjust.log'
+
+handler = logging.FileHandler(LOGFILE)
+FORMAT = "%(asctime)-15s %(levelname)s %(message)s"
+handler.setFormatter(logging.Formatter(FORMAT))
+logger = logging.getLogger(__name__)
+logger.addHandler(handler)
+logger.setLevel(logging.DEBUG)
+
+
+def run_cmd(command):
+    '''
+    run the command and return stdout
+    exit the script if the rc is not zero
+    
+    '''
+    cmdpipe = subprocess.Popen(command, stdout=subprocess.PIPE,
+                               stderr=subprocess.PIPE)
+    stdout, stderr = cmdpipe.communicate()
+    if cmdpipe.returncode != 0:
+        logger.error('%(cmd)s: %(err)s' % {'cmd': ' '.join(command),
+                                           'err': stderr})
+        os._exit(1)
+    return stdout
+
+def increase_pg():
+    if not os.path.isfile(PKLFILE):
+        os._exit(0)
+    with open(PKLFILE, 'rb') as pfd:
+        pgsonpool = pickle.load(pfd)
+    existing_pools = run_cmd(['ceph', 'osd', 'pool', 'ls'])
+    existing_pools = existing_pools.splitlines()
+    # as different pool uses different osds, so we can increase osd numbers
+    # of multi pools at the same time
+    for pool in pgsonpool:
+        if pool not in existing_pools:
+            continue
+        current_pgs = run_cmd(['ceph', 'osd', 'pool', 'get', pool, 'pg_num'])
```

Comment:
All of these commands assume that `ceph` is in $PATH. There probably needs to be an override argument.